near/nep141-locker/src/lib.rs

```diff
@@ -185,6 +206,12 @@ impl FungibleTokenReceiver for Contract {
             NearToken::from_yoctonear(0),
         );
 
+        if self.deployed_tokens.contains(&token_id) {
+            ext_token::ext(token_id.clone())
+                .with_static_gas(BURN_TOKEN_GAS)
+                .burn(amount);
```

Comment:
My original understanding was that we burn tokens only when withdrawing to the original chain, otherwise we lock them just like near-native tokens. Would that be simpler or not?

Reply:
I think if we are transferring from ETH to SOL an erc20 token, then we should consider it as a direct transfer, and the NEAR protocol is used just as a mediator for this transfer.
Also it will be hard to track the total supply in this case, because for now the locked amount should equal to the sum of all total supplies on other chains.